modules/hal_nordic/nrf_802154/CMakeLists.txt

```diff
@@ -5,6 +5,13 @@ zephyr_library_named(nrf-802154-platform)
 zephyr_interface_library_named(zephyr-802154-interface)
 
 if (CONFIG_NRF_802154_RADIO_DRIVER)
+  dt_nodelabel(timer1_node NODELABEL "timer1")
+  dt_node_has_status(status_result PATH ${timer1_node} STATUS okay)
+  if (${status_result})
+	  message(FATAL_ERROR "Resource conflict. IEEE802.15.4 Radio driver requires TIMER1 but "
```

Comment:
I might be wrong, but isn't there a policy that Kconfig/DTS errors must not result in CMake fatal errors, because this effectively locks-out the user from `menuconfig`?
On top of that, catching configuration errors in CMake doesn't seem right. Can it not be somehow achieved using the Kconfig alone?